keps/sig-api-machinery/20191001-oasv3.md

```diff
@@ -0,0 +1,949 @@
+            "type": "string"
+          },
+          "remainingItemCount": {
+            "description": "remainingItemCount is the number of subsequent items in the list which are not included in this list response. If the list request contained label or field selectors, then the number of remaining items is unknown and the field will be left unset and omitted during serialization. If the list is complete (either because it is not chunking or because this is the last chunk), then there are no more remaining items and this field will be left unset and omitted during serialization. Servers older than v1.15 do not set this field. The intended use of the remainingItemCount is *estimating* the size of a collection. Clients should not rely on the remainingItemCount to be set or to be exact.",
+            "type": "integer",
+            "format": "int64"
+          },
+          "resourceVersion": {
+            "description": "String that identifies the server's internal version of this object that can be used by clients to determine when objects have changed. Value must be treated as opaque by clients and passed unmodified back to the server. Populated by the system. Read-only. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#concurrency-control-and-consistency",
+            "type": "string"
+          },
+          "selfLink": {
+            "description": "selfLink is a URL representing this object. Populated by the system. Read-only.\n\nDEPRECATED Kubernetes will stop propagating this field in 1.20 release and the field is planned to be removed in 1.21 release.",
+            "type": "string"
+          }
+        }
+      },
+      "meta.v1.Patch": {
+        "description": "Patch is provided to give a concrete name and type to the Kubernetes PATCH request body.",
+        "type": "object"
+      }
+    },
+    "parameters": {
+      "query.pretty.5ca316":  {
+        "name": "pretty",
+        "in": "query",
+        "description": "If 'true', then the output is pretty printed.",
+        "schema": {
+          "type": "string"
+        }
+      }
+    }
+  }
+}
+```
+
+### New names for objects
+
+The old spec uses the full path to a package as a name for an object, for example `ConfigMap` is defined as `io.k8s.api.core.v1.ConfigMap`.
+This not only doesn't add any value but also reveals the internal path to a type. 
+
+Instead the new names will contain information about a group, version and kind(`GVK`). `ConfigMap` in the new format would become `core.v1.ConfigMap`.
```

Comment:
> Instead the new names will contain information about a group, version and kind(`GVK`). `ConfigMap` in the new format would become `core.v1.ConfigMap`.

for the record, there's no such group as "core", the group name is actually an empty string "" for so-called "core" group.

Reply:
> https://github.com/kubernetes-client/gen/blob/a4d67bd881b30c5f2b1579d38b3d5dfb86e9d086/openapi/preprocess_spec.py#L213-L228. I will double check

it can also be used by third-party libraries i suppose.. we have to preserve backward-compatibilities when refactoring the script.